Update PKGBUILD to reflect new build method

austin-ray opened the pull request:
Updated PKGBUILD to use configure instead of prepare.

Additionally the following changes were introduced:

 * Updated configure options to build identically to community/neomutt
 * Updated depends() to use mailcap instead of mime-types
 * Updated provides() and conflicts() to neomutt
 * Updated .SRCINFO to reflect git changes
 * Update .gitignore to properly ignore subfolders

Lastly, these changes are setup for turning the aur-build repo into the
repo for neomutt-git. flatcap said to push these changes directly to the
branch.

austin-ray committed:
Update PKGBUILD to reflect new build method

Updated PKGBUILD to use configure instead of prepare.

Additionally the following changes were introduced:

  * Updated configure options to build identically to community/neomutt
  * Added docbook-xsl to dependencies. Compilation will fail without.
  * Updated depends() to use mailcap instead of mime-types
  * Updated provides() and conflicts() to neomutt
  * Updated .SRCINFO to reflect git changes
  * Update .gitignore to properly ignore subfolders

Lastly, these changes are setup for turning the aur-build repo into the
repo for neomutt-git. flatcap said to push these changes directly to the
branch.

austin-ray committed:
Update .travis.yml to use different docker image

Previous docker image had become stale and not in neomutt's control,
this uses a fork of that image under neomutt's control.

WhyNotHugo commented on .travis.yml:
```diff
@@ -4,4 +4,4 @@ services:
 
 script:
-  - docker run -v $(pwd):/pkg hobarrera/makepkg
+  - docker run -v $(pwd):/pkg neomutt/aur
```
Uhm... the source for this is hidden, but it _looks_ like a copy-paste of `hobarrera/makepkg`. Would you care to explain that?

austin-ray replied:
The docker image was stale when I tried to build this using hobarrera/makepkg. It refused to recognise a valid dependency. @flatcap said in IRC to fork hobarrera/makepkg into https://github.com/neomutt/aur-docker so rebuilds can be triggered if it goes stale again.

WhyNotHugo replied:
I'm not sure how this was set up, but Docker isn't showing that repository as the source for the `Dockerfile`. In fact, it's not showing any source, or anything. Maybe you used another, private repository, or some other mechanism to build the Docker images?

~In any case, I can set up my `makepkg` repo to trigger builds daily. Since it's used for other packages' CI, I'd prefer to avoid having a duplicate (if possible).~

Updated image to re-build daily. Though it seems that this reference is outdated, and current location [`WhyNotHugo/makepkg`](https://hub.docker.com/r/whynothugo/makepkg/).

WhyNotHugo commented on PKGBUILD:
```diff
@@ -1,56 +1,49 @@
-pkgver=20170912
+_pkgname=neomutt
+pkgname=neomutt-git
+pkgver=neomutt.20171215.r0.gae612851
```
Shouldn't this be `latest`, since the `pkgver` is actually calculated by the function below?

austin-ray replied:
I apologise for missing that before committing. Thank you for catching my mistake.

WhyNotHugo replied:
np, thank you for jumping into this 😄

WhyNotHugo commented on PKGBUILD:
```diff
@@ -1,56 +1,49 @@
-  cd "$pkgname-$pkgver"
+  cd "${srcdir}/${_pkgname}"
+
+  ./configure \
```
Why move this out of `prepare`? It doesn't to actual building, but rather, just prepares the build.

austin-ray replied:
I based this on neomutt-git's PKGBUILD. You make a valid point and I will move it into prepare().

WhyNotHugo commented on .SRCINFO:
```diff
@@ -11,21 +11,18 @@ pkgbase = neomutt
-	sha256sums = 9f0d128fe8dbbbddb67e2b629f8b5e52a9d892f82c735845c1708c14e3046657
+	provides = neomutt
+	conflicts = neomutt
+	source = git+https://github.com/neomutt/neomutt.git#branch=master
```
No need to specify a branch if `master` is to be used.

austin-ray replied:
Duly noted. Will be rectified.

WhyNotHugo reviewed (changes requested):
I've a few reservations about this, mostly the change to a closed-source prebuilt docker image.

austin-ray committed:
Correct PKGBUILD and .SRCINFO

Corrected several errors in PKGBUILD and .SRCINFO. This includes:

  * Setting pkgver=latest. Running makepkg overrides the pkgver and I
    forgot to change pkgver back to latest.
  * configure was moved into prepare()
  * master branch was removed from the git URL

flatcap commented:
@WhyNotHugo 

> I'm not sure how this was set up, but Docker isn't showing that repository as the source for the Dockerfile

That's probably because we've got lots of branches and test repos, atm.
I think all the pieces are in place for a wholly neomutt solution.

> Since it's used for other packages' CI, I'd prefer to avoid having a duplicate (if possible).

Hmm...

I'd rather have a neomutt copy of any repos which we're using for builds, for a couple of reasons:
- Repeatable builds
  I don't want our build changing unless we change it.
- Accessibility 
  Your repo is only accessible by you.
  A neomutt copy would mean members could cooperate on changes. 
  If you were to go on holiday, then we'd need a neomutt copy of the repo if we needed to make changes.

As our _upstream_:
- Any improvements you make, we can pull
- Any changes we make can be turned into PRs

jelly commented on PKGBUILD:
```diff
@@ -1,56 +1,53 @@
+sha256sums=('SKIP')
 arch=('i686' 'x86_64')
-depends=('openssl' 'gdbm' 'mime-types' 'libsasl' 'gnupg' 'gpgme' 'libidn' 'krb5' 'notmuch-runtime')
+depends=('openssl' 'gdbm' 'mailcap' 'libsasl' 'gnupg' 'gpgme' 'libidn' 'krb5' 'notmuch-runtime' 'docbook-xsl')
```
Shouldn't the docbook-xsl not be makedepends?

austin-ray replied:
You are correct. In my haste, I overlooked that particular section. Thank you for noticing my mistake.

jelly commented on PKGBUILD:
```diff
@@ -1,56 +1,53 @@
-depends=('openssl' 'gdbm' 'mime-types' 'libsasl' 'gnupg' 'gpgme' 'libidn' 'krb5' 'notmuch-runtime')
+depends=('openssl' 'gdbm' 'mailcap' 'libsasl' 'gnupg' 'gpgme' 'libidn' 'krb5' 'notmuch-runtime' 'docbook-xsl')
 optdepends=('urlview: for url menu')
 makedepends=('git' 'gnupg' 'libxslt')
```
Should add lynx for generating html documentation to txt.

austin-ray committed:
Update makedepends

Updated makedepends per jelle van der Waa's suggestions.

  * docbook-xsl moved to makedepends
  * lynx added for HTML to txt documentation
